### PR TITLE
Bug-fixes dependencies, included some debugging

### PR DESCRIPTION
**Pull request by carmen-sc**
## Description

- Please include a summary of the change and which issue is fixed.
- Please also include relevant motivation and context.
- List any dependencies that are required for this change.
- Explicitly mention if the current changes can (or could) break any existing interface or functionality.
- If this pull request results in from a fix to a GitHub issue, give a link to it.

## Type of change

Please mark the relevant options with X:

- [x] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change is constrained only to one package, any interfaces to nodes from other packages are maintained
- [ ] This change is constrained only to packages in this repository, any interfaces to nodes from other repositories are maintained
- [ ] This change required a documentation update
- [ ] This change is only a documentation update
- [ ] This change restructures the repository (at level higher than package level)
- [ ] This change requires a change in the Duckiebook (if yes, please describe and, if possible, make a pull request for an edit in the book and link it here)

## Tests

Please describe all tests ran and their outcome. It is highly recommended that you at least do the tests listed by default, even if you only changed the documentation.

- [ ] Made the Duckiebot move with keyboard control
- [ ] Visually checked the published camera stream
- [ ] Verified that the lane following demo works
- [ ] Verified that the indefinite navigation demo works

## Style compliance
Please make sure that all your changes conform to the Duckietown (code style)[link] and code (documentation style)[link] guidelines:

- [x] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I checked that this pull request is compliant with the Duckietown code style
- [ ] I checked that this pull request is compliant with the Duckietown code documentation style

## Next steps
Please describe if you have any questions, comments, or doubts towards the Duckietown software guardians regarding implementation, code style, running tests, documentation, etc.

## FOR THE REVIEWERS
__IMPORTANT:__ DO NOT CHANGE THIS SECTION!

- [ ] Functionality (is the change necessary and is it implemented in the most appropriate way): @liampaull or @afdaniele
- [ ] Integration testing (verify the tests and run additional if necessary): @?
- [ ] Code style compliance: @gibernas
- [ ] Documentation compliance: @aleksandarpetrov

_To the reviewers:_ If you deem necessary, request code edits, more tests, documentation update, or additional reviewing. __Do not review your own code!__

_PR Template, last edit 23 Aug 2019_

### CODE REVIEW

File: packages/camera_driver/src/camera_node.py [hunk 1]
```diff
@@ -8,15 +8,19 @@
...
+    import threading
+    from cv_bridge import CvBridge, CvBridgeError
 
 # for RPi
-if (os.environ.get('ROBOT_HARDWARE') == "raspberry_pi"):
+elif (os.environ.get('ROBOT_HARDWARE') == "raspberry_pi"):
     from picamera import PiCamera
 
+else:
+    raise Exception("Undefined Hardware!")
```
Comment (AleksandarPetrov) [inline packages/camera_driver/src/camera_node.py:22]: "Undefined Hardware!" -> "Undefined Hardware! The environment variable ROBOT_HARDWARE must be set!"

File: packages/wheel_encoder/include/wheel_encoder/wheel_encoder_driver.py [hunk 2]
```diff
@@ -5,9 +5,12 @@
 if (os.environ.get('ROBOT_HARDWARE') == "jetson_nano"):
     import Jetson.GPIO as GPIO
 
-else:
+elif (os.environ.get('ROBOT_HARDWARE') == "raspberry_pi"):
     import RPi.GPIO as GPIO
 
+else:
+    raise Exception("Undefined Hardware!")
```
Comment (AleksandarPetrov) [inline packages/wheel_encoder/include/wheel_encoder/wheel_encoder_driver.py:12]: Same as above

**Review by AleksandarPetrov — CHANGES_REQUESTED** [review]

File: packages/camera_driver/src/camera_node.py [hunk 3]
```diff
@@ -8,15 +8,19 @@
 from threading import Thread
 
 # for Jetson Nano
-import cv2
-import atexit
-import threading
-from cv_bridge import CvBridge, CvBridgeError
+if (os.environ.get('ROBOT_HARDWARE') == "jetson_nano"):
```
Comment (afdaniele) [inline packages/camera_driver/src/camera_node.py:11]: @carmen-sc I introduced this utility function, always prefer Enums to strings
https://github.com/duckietown/dt-commons/blob/1361dc31a0cb7531016ded9de499fc02032499f8/packages/dt_device_utils/device.py#L38

**Review by afdaniele — CHANGES_REQUESTED** [review]

**Comment by liampaull** [comment]
has someone tested this on a Duckiebot?

**Comment by afdaniele** [comment]
@liampaull yes, and it is NOT working. Do not merge yet.

**Pull request merged**